include/swift/Runtime/Mutex.h

```diff
@@ -0,0 +1,276 @@
+//===--- Mutex.h - Lockables ------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// TODO
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_BASIC_MUTEX_H
+#define SWIFT_BASIC_MUTEX_H
+
+#include <functional>
+
+namespace swift {
+
+///
+///
+/// TODO
+///
+///
+class Condition {
```

Comment:
They are useable but holding off on fully switching until I get tests in place, it will allow me to better see the generated code and how it gets optimized.